### ISSUE TITLE
State update endpoint silently discards unused parts of state blob

### ISSUE THREAD

**Issue by lmossman** [airbytehq/airbyte#29639]
## Context

The `/state/create_or_update` API endpoint will discard parts of the state blob which end up being unused, still returning a 200 status code.

For example, if I call that endpoint on a Postgres source connection with the following stream_state body:
```json
[
  {
    "streamDescriptor": {
      "name": "something",
      "an unused field": "the value"
    }
  },
  {
    "streamDescriptor": {
      "name": "purchases",
      "namespace": "public"
    },
    "streamState": {
      "version": 2,
      "state_type": "xmin",
      "num_wraparound": 0,
      "xmin_raw_value": 402793,
      "xmin_xid_value": 402791
    }
  }
]
```
The stream state for `purchases` is valid and used by the endpoint to update the db, but the `something` stream state has an unused field in the descriptor and no `streamState` value set.

In this case, the API happily accepts the request and returns a 200 status code. But what we should probably do is return a special status code (maybe still successful) which indicates that part of the request was thrown away, to make this less confusing for consumers.

## Acceptance Criteria
- The `/state/create_or_update` endpoint returns a special status code when it throws away part of the request
- The Connection state editing UI in the connection Settings page is updated to handle this new response and properly inform the user about the situation

**Comment by josephkmh** [comment]
@lmossman we talked about this in Compose backlog refinement, but none of us really have context on this API or the state management lifecycle (which is more Move's domain). If this is important enough to fix, can connector ops pick it up? If this API is only temporary, maybe it's not worth addressing at all? 🤷

**Comment by lmossman** [comment]
This doesn't feel very high priority to me - it is a bit of a weird UX to add some unsupported object to the array, click save, and then see that object disappear from the editor; but this feels edge-casey enough that I'm okay with us not addressing it unless we see complaints about it.

Does this align with your thinking @evantahler

**Comment by evantahler** [comment]
I agree that this is low-priority.  I would suggest we actually fail (500) the API request in this case with a message if possible ("stream `something` not configured for this connection")
